src/core/lib/iomgr/exec_ctx.h

```diff
@@ -80,10 +80,13 @@ namespace grpc_core {
  *  CONVENTIONS:
  *  - Instance of this must ALWAYS be constructed on the stack, never
  *    heap allocated.
- *  - Exactly one instance of ExecCtx must be created per thread. Instances must
```

Comment:
I don't think we should remove this, since we want to eventually get back to a mode where this is true.  Instead, we should document the fact that we currently have violations of this but that we are going to work toward making those violations go away, and all new code should conform to this restriction to avoid making the problem worse.

Reply:
Shouldn't the comment document reality as it is, and then a TODO should state the desired end-state? I think this has confused several team members who stopped reading at this comment (which is what I did when I implemented the callback API) and don't go on to read the code.

Reply:
I've added further exposition on the TODO below.

Reply:
I specifically want to see this documented outside of the TODO.  The important thing here is that we are telling users about a restriction that they currently need to keep in mind when writing new code, and I think people will not see this if it's buried in a TODO.

Reply:
Ok, I've added it to conventions as something that will change in the future.